mux.c

```diff
@@ -27,6 +27,48 @@
 
 #define BUFSIZE 1024
 
+static ssize_t get(unsigned char *buf, unsigned char *out, size_t len)
+{
+	if (!len)
+		return -1;
+
+	if (buf[0] == IAC) {
+		if (len < 1)
+			return -1;
+		if (buf[1] == IAC) {
+			*out = IAC;
+			return 2;
+		}
+		return -1;
+	} else {
+		*out = buf[0];
+		return 1;
+	}
+}
+
+static size_t getl(unsigned char *buf, uint32_t *out, size_t len)
+{
+	*out = 0;
+	int i;
+	size_t offset = 0;
+
+	for (i = 0; i < 4; ++i) {
+		ssize_t getres;
+		unsigned char c;
+
+		getres = get(buf + offset, &c, len - offset);
+		if (getres < 0)
+			return getres;
+
+		*out << 8;
```

Comment:
This doesn't look right.  You're not actually assigning the result of the shit operation, so out is never actually shifted, and you just end up doing a bitwise and of the values.  I think you want <<= here.

Also, your not using the values anywhere that I can see, but you'll probably need to convert them from network order to host order (see ntohl()), or things like baudrate will be scrambled.

Reply:
> This doesn't look right. You're not actually assigning the result of the shit operation, so out is never actually shifted, and you just end up doing a bitwise and of the values. I think you want <<= here.

Of shift, you're right, force-pushed to use <<=. 

> Also, your not using the values anywhere that I can see, but you'll probably need to convert them from network order to host order (see ntohl()), or things like baudrate will be scrambled.

Yeah, the value isn't used but for the debug printf. I think the endianess handling is fine, as I make sure that the first byte in the buffer it shifted three times by 8 and so is used as the most significant byte in the result. So effectively I opencoded `ntohl()`.